File: BGC_WebApi/BGC_DataAccess/Entities/GameVersion.cs
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BGC_DataAccess.Entities;
public class GameVersion
{
    public int GameVersionId { get; set; }
    
    public int GameId { get; set; } 
    public Game Game { get; set; }

    public ICollection<GameSessionGame> GameSessionGames { get; set; }
    public ICollection<RpgSessionGame> RpgSessionGames { get; set; }
}

public class GameeVersionConfiguration : IEntityTypeConfiguration<GameVersion>
{
    public void Configure(EntityTypeBuilder<GameVersion> builder)
    {
        builder.Property(g => g.Game).IsRequired();
    }
}

File: BGC_Portal/src/app/api/models/game-version.ts
/* tslint:disable */
/* eslint-disable */
import { Game } from './game';
import { GameSessionGame } from './game-session-game';
import { RpgSessionGame } from './rpg-session-game';
export interface GameVersion {
  game?: Game;
  gameId?: number;
  gameSessionGames?: null | Array<GameSessionGame>;
  gameVersionId?: number;
  rpgSessionGames?: null | Array<RpgSessionGame>;
}
